### PR TITLE
fix: generate docs for all features on docs.rs

### PR DESCRIPTION
**Pull request by niklasad1**

### CODE REVIEW

**Review by TarikGul — APPROVED** [review]

File: client/http-client/Cargo.toml [hunk 1]
```diff
@@ -29,3 +29,10 @@ tokio = { version = "1.16", features = ["net", "rt-multi-thread", "macros"] }
 [features]
 default = ["tls"]
 tls = ["hyper-rustls/webpki-tokio"]
+
+[package.metadata.docs.rs]
+all-features = true
+rustdoc-args = ["--cfg", "docsrs"]
```
Comment (jsdw) [inline client/http-client/Cargo.toml:35]: What does this do?

**Review by jsdw — APPROVED** [review]

File: client/http-client/Cargo.toml (hunk 1 above)
Comment (niklasad1) [reply client/http-client/Cargo.toml:35]: By default `docs.rs` only builds the docs with default-features so we want all features to be documented.

Comment (jsdw) [reply client/http-client/Cargo.toml:35]: Ah thanks, and the `--cfg docsrs` bit is about making it possible to conditionally compile stuff for doc purposes I gather!

Comment (niklasad1) [reply client/http-client/Cargo.toml:35]: should be equivalent to `RUSTDOCFLAGS="--cfg docrs" cargo doc --workspace --no-deps --document-private-items` it seems.

**Pull request merged**